src/nemos/basis/_basis.py

```diff
@@ -524,6 +697,19 @@ def __pow__(self, exponent: int) -> MultiplicativeBasis:
     def __repr__(self):
         return format_repr(self)
 
+    def __getitem__(self, index: str) -> Basis:
+
+        search = next(
+            (bas for lab, bas in generate_basis_label_pair(self) if lab == index), None
+        )
+
+        if search is None:
+            avail_index = ",".join(f"'{b}'" for b in self._generate_subtree_labels())
+            raise IndexError(
+                f"Basis label {index} not found. Available labels: {avail_index}"
```

Comment:
maybe have a different error message if `index` is an int? because then someone just assumed they can do index-based getting, whereas we only support label-based getting.